server/src/api/taskratings/taskratings.controller.ts

```diff
@@ -186,13 +186,23 @@ export const patchTaskratings: RouteHandlerFnc = async (ctx) => {
   if (foundRatings.some(({ err }) => err === 403))
     return void (ctx.status = 403);
 
-  const updated = await Taskrating.transaction((trx) =>
-    Promise.all(
+  const result = await Taskrating.transaction(async (trx) => {
+    const toDelete = foundRatings.filter(({ value }) => !value);
+    const deleted = await Promise.all(
+      toDelete.flatMap(({ rating }) =>
+        rating ? Taskrating.query(trx).findById(rating.id).delete() : [],
+      ),
+    );
+
+    if (deleted.length !== toDelete.length) throw new Error();
```

Comment:
Just want to highlight this, as it might be questionable. I wanted to rollback the transaction in case some of the deletions are not made, hence the error. I am not sure though if it is required and whether to add an error description (string)

Reply:
I suppose this could happen if more than one user remove the same rating at the same time.
I don't think that should be necessarily be an error, since the deletion failed because the item was already gone.